reactive/kubernetes_master.py

```diff
@@ -1036,21 +987,23 @@ def start_master():
     configure_scheduler()
 
     # kube-proxy
-    cni = endpoint_from_flag('cni.available')
-    default_cni = hookenv.config('default-cni')
-    cluster_cidr = cni.get_config(default=default_cni)['cidr']
-    # Set bind address to work around node IP error when there's no kubelet
-    # https://bugs.launchpad.net/charm-kubernetes-master/+bug/1841114
-    bind_address = get_ingress_address('kube-control')
+    cluster_cidr = kubernetes_common.cluster_cidr()
+    if kubernetes_common.is_ipv6(cluster_cidr):
+        kubernetes_common.enable_ipv6_forwarding()
     configure_kube_proxy(configure_prefix,
-                         ['127.0.0.1:8080'], cluster_cidr,
-                         bind_address=bind_address)
+                         ['127.0.0.1:8080'], cluster_cidr)
     service_restart('snap.kube-proxy.daemon')
 
     set_state('kubernetes-master.components.started')
     hookenv.open_port(6443)
 
 
+@when('config.changed.proxy-extra-args')
+def proxy_args_changed():
+    clear_flag('kubernetes-master.components.started')
+    clear_flag('config.changed.proxy-extra-args')
```

Comment:
This change fixes https://bugs.launchpad.net/charm-kubernetes-master/+bug/1841112. I've linked this PR to the issue and assigned you accordingly.